### ISSUE TITLE
Create an integration with Curate

### ISSUE THREAD

**Issue by erussey** [emory-libraries/aspace#10]
Information to retrieve from the API (below includes examples from the William Levi Dawson papers":

- **Title**, repository record, "title":"William Levi Dawson papers"
William Levi Dawson papers

- **Description/Abstract**, resource record, "notes":[{"jsonmodel_type":"note_singlepart","content":["Papers of William Levi Dawson, African American composer, conductor, and educator from Anniston, Alabama, including correspondence, original scores of Dawson's works, personal and family papers, photographs, audio visual materials, and printed material."],"type":"abstract","persistent_id":"64080a4d3ea0603ef38de4b28f34312e","publish":true}
Papers of William Levi Dawson, African American composer, conductor, and educator from Anniston, Alabama, including correspondence, original scores of Dawson's works, personal and family papers, photographs, audio visual materials, and printed material.

- **Administrative Unit**, repository record, in the subrecord under "agent_representation":{"ref":}

- **Institution**: repository record, "parent_institution_name":"Emory University Libraries"
Emory University Libraries

- **Holding Repository**, repository record, in the subrecord under "agent_representation":{"ref":}

- **Creator** (**repeatable**), resource record, in the subrecord under "linked_agents":[{"role":"creator","terms":[],"ref":"/agents/people/1891"}

- **System of Record ID** 
*May need to be constructed from the resource record identifier.
Resource record identifier: https://emory-dev.lyrasistechnology.org/staff/resources/1037
PUI identifier: https://emory-dev.lyrasistechnology.org/repositories/7/resources/1037

- **Call number**, resource record "id_0":"Manuscript Collection No. 892"

- **Primary language**, resource record "jsonmodel_type":"lang_material","notes":[],"language_and_script":{"lock_version":0,"create_time":"2019-10-30T22:44:30Z","system_mtime":"2019-10-30T22:44:30Z","user_mtime":"2019-10-30T22:44:30Z","language":"eng","jsonmodel_type":"language_and_script"}}]
***language codes will need to be transformed into human readable forms (eng=english)**

- **Keywords** (**repeatable**)

- **Subject-Topics** (**repeatable**), resource record in the subrecord under "subjects":[{"ref":"/subjects/2749"},{"ref":"/subjects/2750"},..

- **Subject-Names** (**repeatable**), resource record in the subrecord under "linked_agents":[{"role":"subject","terms":[],"ref":"/agents/people/1892"},{"role":"subject","terms":[],"ref":"/agents/people/1893"},{"role":"subject","terms":[],"ref":"/agents/people/6403"},{"role":"subject","terms":[],"ref":"/agents/people/1895"},{"role":"subject","terms":[],"ref":"/agents/people/1896"},

- **Subject-Geographic** (**repeatable**) resource record in the subrecord under "linked_agents"{"ref":"/subjects/2754"}

- **Subject-Time Periods** (**repeatable**)
****need to find an example for this one.**

- **Repository contact information**, repository record, in the subrecord under "agent_representation":{"ref": }

**Comment by erussey** [comment]
I spoke with @eporter23 about this ticket in January.  We agreed to scope this work as part of the ASpace project although the application that will need code development is curate.

**Comment by erussey** [comment]
For testing:
ASpace identifier for Rose in the dev environment: https://emory-dev.lyrasistechnology.org/staff/repositories/7
ASpace identifier for the William Levi Dawson papers: https://emory-dev.lyrasistechnology.org/staff/resources/1037 

If you send me the API output for the collection, I can work on mapping those fields to the Curate fields above.  I think that we'll have to look at subrecords for some of the fields.

**Comment by erussey** [comment]
Attaching Curate controlled vocabulary lists for reference.  The values for the Holding Repository and Administrative Unit will have to match the values in this list: https://docs.google.com/spreadsheets/d/1tm7RDYWVcAWEZuA-lB3cbxxGI4qjn8uCn71X2EJPB50/edit#gid=854442392 .  I have added a value to ASpace for the holding repository that we can pull from the "agent_representation" record...we may have to do a mapping as part of the API script for Administrative Unit.

**Comment by erussey** [comment]
@abelemlih : I've added more information to the mappings in the ticket.  I do need some subrecords to be able to give you the full mapping.  See Administrative Unit, Holding repository, creator, all subjects, and repository contact information.

**Comment by erussey** [comment]
For notekeeping purposes, the sample JSON API response for a repository record:
`{"lock_version":2,"repo_code":"Rose","name":"Stuart A. Rose Manuscript, Archives, and Rare Book Library","parent_institution_name":"Emory University Libraries","url":"https://libraries.emory.edu/rose","created_by":"admin","last_modified_by":"avery.j.johnson","create_time":"2018-07-05T12:59:20Z","system_mtime":"2023-02-01T21:05:27Z","user_mtime":"2023-02-01T21:05:27Z","publish":true,"oai_is_disabled":false,"oai_sets_available":"[\"888\",\"889\",\"890\",\"891\",\"892\",\"893\",\"894\",\"895\",\"896\",\"897\",\"898\"]","slug":"rose_1_1","is_slug_auto":false,"country":"US","jsonmodel_type":"repository","uri":"/repositories/7","display_string":"Stuart A. Rose Manuscript, Archives, and Rare Book Library (Rose)","agent_representation":{"ref":"/agents/corporate_entities/557"}}`

And a resource record:
`{"lock_version":1,"title":"William Levi Dawson papers","publish":true,"restrictions":false,"ead_id":"dawson892","ead_location":"http://pid.emory.edu/ark:/25593/8z6qq","finding_aid_title":"William Levi Dawson papers, 1903-1990","finding_aid_date":"July 24, 2006","finding_aid_author":"Stuart A. Rose Manuscript, Archives, and Rare Book Library, Emory University","finding_aid_language_note":"eng","created_by":"admin","last_modified_by":"admin","create_time":"2018-09-06T17:05:39Z","system_mtime":"2022-01-28T14:35:09Z","user_mtime":"2018-09-06T17:05:39Z","suppressed":false,"is_slug_auto":false,"id_0":"Manuscript Collection No. 892","level":"collection","finding_aid_language":"und","finding_aid_script":"Zyyy","jsonmodel_type":"resource","external_ids":[],"subjects":[{"ref":"/subjects/2749"},{"ref":"/subjects/2750"},{"ref":"/subjects/1788"},{"ref":"/subjects/1145"},{"ref":"/subjects/2751"},{"ref":"/subjects/1891"},{"ref":"/subjects/1322"},{"ref":"/subjects/1892"},{"ref":"/subjects/1800"},{"ref":"/subjects/2752"},{"ref":"/subjects/2753"},{"ref":"/subjects/931"},{"ref":"/subjects/2754"},{"ref":"/subjects/2386"},{"ref":"/subjects/391"},{"ref":"/subjects/448"},{"ref":"/subjects/1790"},{"ref":"/subjects/1356"},{"ref":"/subjects/753"},{"ref":"/subjects/2755"},{"ref":"/subjects/2756"}],"linked_events":[],"extents":[{"lock_version":0,"number":"70.25","container_summary":"(120 boxes); 11 oversized papers boxes and 2 oversized papers folders (OP)","created_by":"admin","last_modified_by":"admin","create_time":"2018-09-06T17:05:39Z","system_mtime":"2018-11-05T17:44:23Z","user_mtime":"2018-09-06T17:05:39Z","portion":"whole","extent_type":"linear_feet","jsonmodel_type":"extent"}],"lang_materials":[{"lock_version":0,"create_time":"2018-09-06T17:05:39Z","system_mtime":"2018-09-06T17:05:39Z","user_mtime":"2018-09-06T17:05:39Z","jsonmodel_type":"lang_material","notes":[],"language_and_script":{"lock_version":0,"create_time":"2018-09-06T17:05:39Z","system_mtime":"2018-09-06T17:05:39Z","user_mtime":"2018-09-06T17:05:39Z","language":"eng","jsonmodel_type":"language_and_script"}},{"lock_version":0,"create_time":"2018-09-06T17:05:39Z","system_mtime":"2018-09-06T17:05:39Z","user_mtime":"2018-09-06T17:05:39Z","jsonmodel_type":"lang_material","notes":[{"jsonmodel_type":"note_langmaterial","content":["Materials mostly in English."],"type":"langmaterial","persistent_id":"13bb4f77d529937907b17bc48915e3f3","publish":true}]},{"lock_version":0,"create_time":"2019-10-30T22:44:30Z","system_mtime":"2019-10-30T22:44:30Z","user_mtime":"2019-10-30T22:44:30Z","jsonmodel_type":"lang_material","notes":[],"language_and_script":{"lock_version":0,"create_time":"2019-10-30T22:44:30Z","system_mtime":"2019-10-30T22:44:30Z","user_mtime":"2019-10-30T22:44:30Z","language":"eng","jsonmodel_type":"language_and_script"}}],"dates":[{"lock_version":0,"expression":"1903-1990","begin":"1903","end":"1990","created_by":"admin","last_modified_by":"admin","create_time":"2018-09-06T17:05:39Z","system_mtime":"2018-09-06T17:05:39Z","user_mtime":"2018-09-06T17:05:39Z","date_type":"inclusive","label":"creation","era":"ce","calendar":"gregorian","jsonmodel_type":"date"}],"external_documents":[],"rights_statements":[],"linked_agents":[{"role":"creator","terms":[],"ref":"/agents/people/1891"},{"role":"subject","terms":[],"ref":"/agents/people/1892"},{"role":"subject","terms":[],"ref":"/agents/people/1893"},{"role":"subject","terms":[],"ref":"/agents/people/6403"},{"role":"subject","terms":[],"ref":"/agents/people/1895"},{"role":"subject","terms":[],"ref":"/agents/people/1896"},{"role":"subject","terms":[],"ref":"/agents/corporate_entities/1059"},{"role":"subject","terms":[],"ref":"/agents/corporate_entities/1060"},{"role":"subject","terms":[],"ref":"/agents/corporate_entities/1061"},{"role":"subject","terms":[],"ref":"/agents/corporate_entities/1062"},{"role":"subject","terms":[],"ref":"/agents/corporate_entities/1063"}],"import_previous_arks":[],"revision_statements":[{"date":"July 29, 2015","description":"Finding aid revised to include enhanced metadata tagging.","created_by":"admin","last_modified_by":"admin","create_time":"2018-09-06T17:05:39Z","system_mtime":"2019-06-14T21:23:49Z","user_mtime":"2018-09-06T17:05:39Z","publish":false,"jsonmodel_type":"revision_statement","uri":"/revision_statement/98","repository":{"ref":"/repositories/7"}}],"instances":[],"deaccessions":[],"related_accessions":[{"ref":"/repositories/7/accessions/2434"},{"ref":"/repositories/7/accessions/6055"},{"ref":"/repositories/7/accessions/2750"},{"ref":"/repositories/7/accessions/2890"},{"ref":"/repositories/7/accessions/2891"}],"classifications":[],"notes":[{"jsonmodel_type":"note_singlepart","content":["Papers of William Levi Dawson, African American composer, conductor, and educator from Anniston, Alabama, including correspondence, original scores of Dawson's works, personal and family papers, photographs, audio visual materials, and printed material."],"type":"abstract","persistent_id":"64080a4d3ea0603ef38de4b28f34312e","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Gift, 2001"}],"type":"acqinfo","label":"Source","persistent_id":"ccd8c371d60b3ffb4c180422961dbb01","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Special restrictions apply: Use copies have not been made for all of the audiovisual series at this time. Researchers must contact the Rose Library in advance for access to these materials.\n\n Collection stored off-site. Researchers must contact the Rose Library in advance to access this collection."}],"type":"accessrestrict","label":"Restrictions on Access","persistent_id":"1446481d00832c7a12086c6b2950d254","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Printed or manuscript music in this collection that is still under copyright protection and is not in the Public Domain may not be photocopied or photographed. Researchers must provide written authorization from the copyright holder to request copies of these materials."}],"type":"userestrict","label":"Terms Governing Use and Reproduction","persistent_id":"f1d5d8cb230a0661de7adec30fd69634","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"[after identification of item(s)], William Levi Dawson papers, Stuart A. Rose Manuscript, Archives, and Rare Book Library, Emory University."}],"type":"prefercite","label":"Citation","persistent_id":"871aeec7acf303b5d8510b698501167d","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Processed by Michelle Hite and Elizabeth Russey, October 20, 2005.\n\n <note> Control level 3\n\n </note> <note> File name: dawson892.doc\n\n </note>"}],"type":"processinfo","label":"Processing","persistent_id":"daa55c588a40ab99d35d1e4740033eb2","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Control level 3"}],"type":"odd","persistent_id":"a48b9aedf6286af73fa3da053c9411e0","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"File name: dawson892.doc"}],"type":"odd","persistent_id":"dacd4e0767c4eb15f403433f3daa8310","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"William Levi Dawson (1899-1990), African American composer, conductor, and educator, was born in <geogname source=\"geonames\" authfilenumber=\"4830198\">Anniston, Alabama</geogname>, the oldest of the seven children of <persname>George W. Dawson</persname>, an illiterate day laborer and former slave, and <persname>Eliza Starkey Dawson</persname>. At the age of thirteen, he ran away from home to attend <corpname source=\"viaf\" authfilenumber=\"150721832\">Tuskegee Institute</corpname>, earning tuition by working in the Agricultural Division for the next seven years. He was admitted to the Institute band and orchestra, under the direction of <persname>Frank L. Drye</persname>, and learned to play most of the instruments. He joined the <corpname source=\"viaf\" authfilenumber=\"130256235\">Institute Choir</corpname>, under Jennie Cheatham Lee, and traveled extensively with the <corpname source=\"viaf\" authfilenumber=\"143248742\">Tuskegee Singers</corpname> and with the Institute band and orchestra. Graduating in 1921, he studied composition and orchestration with <persname>Henry V. Stearns</persname> at <corpname source=\"viaf\" authfilenumber=\"147866447\">Washburn College</corpname> in <geogname source=\"geonames\" authfilenumber=\"4280539\">Topeka, Kansas</geogname>. He also spent four years in the study of theory and counterpoint with <persname>Regina G. Hall</persname> and Dr. <persname source=\"viaf\" authfilenumber=\"69701531\">Carl Busch</persname> at the <corpname>Horner Institute of Fine Arts</corpname> in <geogname source=\"geonames\" authfilenumber=\"4393217\">Kansas City, Missouri</geogname>, graduating in 1925 with a Bachelor of Music degree. In 1927 he received a Master of Music degree in composition from the <corpname source=\"viaf\" authfilenumber=\"155974805\">American Conservatory of Music</corpname> in <geogname source=\"geonames\" authfilenumber=\"4887398\">Chicago</geogname>.\n\n In 1921, Dawson became director of music at <corpname>Kansas Vocational College</corpname> in Topeka. The next year he became director of music at <corpname source=\"viaf\" authfilenumber=\"128203694\">Lincoln High School</corpname>, Kansas City, Missouri. While studying in Chicago, he became the director of one of the principal church choirs of that city and played first trombone in the <corpname source=\"viaf\" authfilenumber=\"156127913\">Chicago Civic Orchestra</corpname>. In 1927 he married <persname>Cornella Lampton</persname>, who died less than a year later. He would marry again on September 1, 1935 to <persname source=\"viaf\" authfilenumber=\"66230210\">Cecile Demae Nicholson</persname> in <geogname source=\"geonames\" authfilenumber=\"4180439\">Atlanta, Georgia</geogname>.\n\n In the fall of 1930, he was invited to return to Tuskegee Institute, to organize and conduct its School of Music. In addition to his administrative duties, Dawson conducted the Tuskegee Institute Choir. Under his leadership, the choir performed at the opening of <corpname source=\"viaf\" authfilenumber=\"151422379\">Radio City Music Hall</corpname> in <geogname source=\"geonames\" authfilenumber=\"5128581\">New York</geogname> in 1932, sang for both <persname source=\"viaf\" authfilenumber=\"14778660\">Herbert Hoover</persname> and <persname source=\"viaf\" authfilenumber=\"29542759\">Franklin D. Roosevelt</persname>, and was the first African American performing organization to appear at <corpname source=\"viaf\" authfilenumber=\"316593475\">Constitution Hall</corpname> in <geogname source=\"geonames\" authfilenumber=\"4140963\">Washington, D.C.</geogname>, in 1946, breaking a long-standing race barrier. Despite the busy touring schedule of the Tuskegee Institute Choir, Dawson frequently traveled internationally. In 1952 he took a year-long sabbatical to West Africa, visiting <geogname source=\"geonames\" authfilenumber=\"2403846\">Sierra Leone</geogname>, <geogname source=\"geonames\" authfilenumber=\"2275384\">Liberia</geogname>, the Gold Coast (now <geogname source=\"geonames\" authfilenumber=\"2300660\">Ghana</geogname>), <geogname source=\"geonames\" authfilenumber=\"2328926\">Nigeria</geogname>, <geogname source=\"geonames\" authfilenumber=\"2245662\">Senegal</geogname>, and Dahomey (now <geogname source=\"geonames\" authfilenumber=\"2395170\">Benin</geogname>). In 1956 the <corpname source=\"viaf\" authfilenumber=\"125408196\">United States State Department</corpname> invited Dawson to tour <geogname source=\"geonames\" authfilenumber=\"2510769\">Spain</geogname> to train local choirs in the African American spiritual tradition. That same year he retired from the Tuskegee Institute, after 25 years of leadership in the music department.\n\n A prolific composer and arranger, Dawson's original compositions include <title source=\"oclc\" authfilenumber=\"497423584\" type=\"songs\">Forever Thine</title> (1920), <title source=\"oclc\" authfilenumber=\"22669478\" type=\"songs\">Out in the Fields</title> (1930), and, most famously, the <title source=\"oclc\" authfilenumber=\"2036546\" type=\"songs\">Negro Folk Symphony</title>, premiered in 1934 by the <corpname source=\"viaf\" authfilenumber=\"149065320\">Philadelphia Orchestra</corpname> under the direction of <persname source=\"viaf\" authfilenumber=\"24788527\">Leopold Stokowski</persname>. During his tenure as the director of the Tuskegee Choir, Dawson also composed a number of arrangements of African American spirituals such as \"King Jesus is a-Listening,\" \"There Is a Balm in Gilead,\" and \"Ezekiel Saw de Wheel.\" He also established his own music publishing business, printing his arrangements under the imprint <corpname>Music Press</corpname>.\n\n William Levi Dawson died May 2, 1990, in <geogname source=\"geonames\" authfilenumber=\"4076784\">Montgomery, Alabama</geogname>, at the age of 90."}],"type":"bioghist","label":"Biographical Note","persistent_id":"04265cc54a073a0b6ac9839f2367da38","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"The collection contains the personal papers of William Levi Dawson from 1903-1990. The papers include correspondence, original scores of Dawson's works; files relating to Dawson's music publishing; writings by Dawson; subject files; notebooks, address books; and scrapbooks; personal and family papers; photographs; audio visual materials; ephemera; and printed material."}],"type":"scopecontent","label":"Scope and Content Note","persistent_id":"c484a08d38153dad8799af9cfd0fdb49","publish":true},{"jsonmodel_type":"note_multipart","subnotes":[{"publish":true,"jsonmodel_type":"note_text","content":"Organized into twelve series: (1) Correspondence, (2) Scores, (3) Music publishing files, (4) Writings by Dawson, (5) Subject files, (6) Notebooks, address books, scrapbooks, (7) Other personal and family papers, (8) Photographs, (9) Printed material, (10) Ephemera, (11) Audio-visual materials, and (12) Collected material."}],"type":"arrangement","label":"Arrangement Note","persistent_id":"7b0b310367eaba337197a2afba4b4c56","publish":true}],"metadata_rights_declarations":[],"uri":"/repositories/7/resources/1037","repository":{"ref":"/repositories/7"},"tree":{"ref":"/repositories/7/resources/1037/tree"}}`

**Comment by erussey** [comment]
Finding aid with Geographic subject heading: Civil War Collection (Rose repository, resource 5598)
I couldn't find any temporal headings...could you search the API and see if you can find a subject with type "temporal"?

**Comment by erussey** [comment]
Resources to test API script with: 
Pitts: 7144
Rose: 6552, 6401, 6569
Law: 4379
Health: 5328
EUA: 7240

Please send me the output for each of these.

**Comment by erussey** [comment]
@abelemlih : seeing a couple of things.  

- Repositories other than Rose and EUA do not have an administrative unit; the logic we created to generate that information only should apply to the Rose and EUA repositories.

- The :institution should *always be Emory University, so while we could map from ASpace, I think it's better for this field to be a constant value

- We are going to have to create a mapping for the holding repository.  Curate has a controlled value list, so the values you're pulling out of ASpace are correct, but need to show up in Curate as a matched value from what is in the API extract in order to meet the requirements of the Curate application.  I can create a mapping table for you for this.

**Comment by abelemlih** [comment]
@erussey I edited how the institution is set in the repository data, and it will now always be the constant value 'Emory University'. The following is data for the Rose repository:

```
{:name=>"Stuart A. Rose Manuscript, Archives, and Rare Book Library", :administrative_unit=>"Stuart A. Rose Manuscript, Archives, and Rare Book Library", :holding_repository=>"Stuart A. Rose Manuscript, Archives, and Rare Book Library", :institution=>"Emory University", :contact_information=>"Stuart A. Rose Manuscript, Archives, and Rare Book Library\nEmory University Libraries\n540 Asbury Circle\nAtlanta, Georgia, 30322\nrose.library@emory.edu\n404-727-6887"} 
```

As for the two other comments, those are changes we will need to implement in the backend portion of the next ticket, #85. As we discussed earlier regarding mapping language (e.g. "eng" to "English"), in that ticket, I will add all mapping logic that ensures only data that matches Curate's business logic is autopopulated. The reason I am preserving the ArchivesSpace data as is in this service is to ensure when debugging, all data the service generates actually came from ArchivesSpace, and was not modified through custom business logic along the way.

If all issues are addressed, I will go ahead and get the PR merged, and we can close this ticket. Let me know if you have any questions!
